backend/controller/user.go

```diff
@@ -2,14 +2,87 @@ package controller
 
 import (
 	"api/config"
+	"api/helpers"
 	"api/models"
 	"encoding/json"
+	"github.com/go-chi/chi/v5"
+	"log"
 	"net/http"
+	"strconv"
 )
 
 func GetAllUsers(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
 	users := []models.User{}
-	config.DB.Find(&users)
+	result := config.DB.Find(&users)
+	if result.Error != nil {
+		helpers.DBErrorHandling(result.Error, w, r)
+		return
+	}
+	//go yells at you if you don't error check this, better to be safe
+	errJson := json.NewEncoder(w).Encode(&users)
+	if errJson != nil {
+		http.Error(w, "500", http.StatusInternalServerError)
+		return
+	}
+}
+func GetUser(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	uId := chi.URLParam(r, "userId")
+	var user models.User
+	result := config.DB.First(&user, uId)
+
+	if result.Error != nil {
+		helpers.DBErrorHandling(result.Error, w, r)
+		return
+	}
+	errJson := json.NewEncoder(w).Encode(&user)
+	if errJson != nil {
+		http.Error(w, "500", http.StatusInternalServerError)
+		return
+	}
+}
+
+func CreateUser(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	var user models.UserReq
```

Comment:
What is UserReq vs. User?

Reply:
Its in models, essentially userReq contains a list of all the parameters that can be inputted to, (name, email, IsAdmin) and not the non-inputtable params, this way during user input we can verify against the proper params